crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -314,6 +314,8 @@ void EVP_Cipher_verify_service_indicator(const EVP_CIPHER_CTX *ctx) {
     case NID_aes_128_ctr:
     case NID_aes_192_ctr:
     case NID_aes_256_ctr:
+
+    case NID_aes_256_xts:
```

Comment:
`EVP_Cipher_verify_service_indicator()` is called within `EVP_Cipher()`, `EVP_EncryptFinal_ex()` and `EVP_DecryptFinal_ex()`. However, in the case of XTS, the `Final` functions are NOP, will we check in `Update` whether the cipher is xts to call `EVP_Cipher_verify_service_indicator`? Note also that in the case of xts, update can be called only once after init, it cannot be called repeatedly; it does the whole encryption in one shot to do the cipher stealing at the end.
I don't know if all the other ciphers listed here need to call final or not, but I guess yes.

Reply:
In my opinion, I would keep the original service indicator implementation of leaving the checks within the `*Final` operations. `*Final` might be a no-op for XTS, but it was within our general service indicator usage guidance that this "*Final" step was our way of differentiating if the user had finished the "FIPS service". This general guidance goes for all the crypto functions within AWS-LC.
If we tried to add a check within `EVP_Update`, it would add additional complexity for us to maintain this outlier and our service indicator usage guidance would have to be updated as well.